src/v/iceberg/tests/merge_append_action_test.cc

```diff
@@ -635,3 +638,119 @@ TEST_F(MergeAppendActionTest, TestTagWithExpiration) {
     ASSERT_TRUE(tag_snap.max_ref_age_ms.has_value());
     ASSERT_EQ(long_max, tag_snap.max_ref_age_ms.value());
 }
+
+TEST_F(MergeAppendActionTest, TestMultiplePartitionSpecs) {
+    const size_t num_to_merge_at
+      = merge_append_action::default_min_to_merge_new_files;
+    const size_t files_per_man = 2;
+    const size_t rows_per_man = 25;
+    transaction tx(create_table());
+
+    auto files_with_first_spec = [&]() {
+        return create_data_files(
+          tx.table(),
+          "foo",
+          files_per_man,
+          rows_per_man,
+          int_value{42},
+          tx.table().schemas.at(0).schema_id,
+          tx.table().partition_specs.at(0).spec_id);
+    };
+
+    // Repeatedly add 50 manifests with old partition spec.
+    for (size_t i = 0; i < num_to_merge_at / 2; i++) {
+        const auto expected_snapshots = i + 1;
+        const auto expected_manifests = expected_snapshots;
+        merge_append_and_check(
+          tx, files_with_first_spec(), expected_snapshots, expected_manifests);
+    }
+
+    // Add new partition spec and make it default.
+    {
+        auto table = std::move(tx).release_metadata();
+        auto new_spec = partition_spec{.spec_id = partition_spec::id_t{1}};
+        new_spec.fields.push_back(partition_field{
+          .source_id = nested_field::id_t{3}, // baz
+          .field_id = partition_field::id_t{1001},
+          .name = "baz",
+          .transform = identity_transform{},
+        });
+        table.partition_specs.push_back(std::move(new_spec));
+        table.default_spec_id = table.partition_specs.back().spec_id;
+        tx = transaction(std::move(table));
+    }
+
+    auto files_with_second_spec = [&]() {
+        return create_data_files(
+          tx.table(),
+          "foo",
+          files_per_man,
+          rows_per_man,
+          boolean_value{true},
+          tx.table().schemas.at(0).schema_id,
+          tx.table().partition_specs.at(1).spec_id);
```

Comment:
nitpick: could use `default_spec_id` here, no?

Reply:
Yes, they should be the same. Though I don't feel that adds much to readability TBH

Reply:
fair enough